### DISCUSSION TITLE
Compiling error - isysroot - pyhit

### DISCUSSION THREAD

**Discussion by favinom** [idaholab/moose#16415] [Q&A Getting Started] [answered]
I am installing the latest moose, using manual install and I got

Building and linking /Users/favinom/projects/moose/framework/../python/pyhit/hit.so...
g++: error: unrecognized command line option '-iwithsysroot'; did you mean '-isysroot'?
g++: error: unrecognized command line option '-iwithsysroot'; did you mean '-isysroot'?
make: *** [/Users/favinom/projects/moose/framework/../python/pyhit/hit.so] Error 1

Does anyone know how to fix this?

**Comment by ngrilli** [comment, marked as answer]
I have also got problem with pyhit during installation on cluster.
I had to install pyhit manually to fix it.

At which point of the installation procedure you get this error?
What is your operating system?
Did you install yourself compiler and mpich?

**Reply by favinom** [reply]
Ciao,

I am getting this problem when I am in the test folder and try to compile.
I am on osx, using gcc from hpc.sourceforge, mpich compiled myself

**Reply by GiudGiud** [reply]
Did this resolve the issue?

**Reply by favinom** [reply]
EDITED

FIXED: read below

The error appears when the compiler does

cd "[]/moose/framework/contrib/hit"

and then

mpicxx -std=c++11 -w -fPIC -lstdc++ -shared \
/Users/favinom/projects/real/moose/framework/contrib/hit/hit.cpp \
/Users/favinom/projects/real/moose/framework/contrib/hit/lex.cc \
/Users/favinom/projects/real/moose/framework/contrib/hit/parse.cc \
/Users/favinom/projects/real/moose/framework/contrib/hit/braceexpr.cc \
-L/System/Library/Frameworks/Python.framework/Versions/2.7/lib \
-iwithsysroot /System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 \
-iwithsysroot /System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 \
-undefined dynamic_lookup \
-o []/moose/framework/../python/pyhit/hit.so

It seems that -iwithsysroot is not an option of my mpicxx.
Actually, the only option that I see running "mpicxx --help" is

--sysroot=<directory>

but if I try to run

mpicxx -std=c++11 -w -fPIC -lstdc++ -shared \
/Users/favinom/projects/real/moose/framework/contrib/hit/hit.cpp \
/Users/favinom/projects/real/moose/framework/contrib/hit/lex.cc \
/Users/favinom/projects/real/moose/framework/contrib/hit/parse.cc \
/Users/favinom/projects/real/moose/framework/contrib/hit/braceexpr.cc \
-L/System/Library/Frameworks/Python.framework/Versions/2.7/lib \
--sysroot='/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7' \
-undefined dynamic_lookup \
-o []/moose/framework/../python/pyhit/hit.so

some inclusions fail. So I tried 

mpicxx -std=c++11 -w -fPIC -lstdc++ -shared \
/Users/favinom/projects/real/moose/framework/contrib/hit/hit.cpp \
/Users/favinom/projects/real/moose/framework/contrib/hit/lex.cc \
/Users/favinom/projects/real/moose/framework/contrib/hit/parse.cc \
/Users/favinom/projects/real/moose/framework/contrib/hit/braceexpr.cc \
-L/System/Library/Frameworks/Python.framework/Versions/2.7/lib \
-I/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 \
-undefined dynamic_lookup \
-o []/moose/framework/../python/pyhit/hit.so

and it compiled the library. I went back to the 'test' folder and proceeded to compile the rest.

replace [] with the correct folder in your system.
